docs/how_to_guides/creating_and_editing_expectations/how_to_create_custom_expectations_for_sqlalchemy.rst

```diff
@@ -3,16 +3,266 @@
+
+    |
+
+    You can also define ``column_aggregate_expectations``, which generate a single observed value for a whole column and evaluate it against some criteria. Like ``column_map_expectations``, ``column_aggregate_expectations`` also take at least two arguments (``self`` and ``column``). In this case, it's your responsibility to create all the logic for the full query, then wrap the result in a return object with a Boolean ``success`` value and ``observed_value`` nested inside ``result``.
+    
+    .. code-block:: python
+
+        @MetaSqlAlchemyDataset.column_aggregate_expectation
+        def expect_column_mode_to_equal(self, column, value):
+            mode_query = sa.select([
+                sa.column(column).label('value'),
+                sa.func.count(sa.column(column)).label('frequency')
+            ]).select_from(self._table).group_by(sa.column(column)).order_by(sa.desc(sa.column('frequency')))
+
+            mode = self.engine.execute(mode_query).scalar()
+            return {
+                "success": mode == value,
+                "result": {
+                    "observed_value": mode,
+                }
+            }
+
+    By convention, ``column_aggregate_expectations`` always start with ``expect_column_{property}_...`` (Ex: ``expect_column_mean_to_be_between``, ``expect_column_most_common_value_to_be_in_set``, ``expect_column_chisquare_test_p_value_to_be_greater_than``). Following this pattern is highly recommended, but not strictly required. If you want to confuse yourself with bad names, the package won't stop you.
+
+    |
+
+    Please see the :ref:`SqlAlchemyDataset` module for additional examples of how to contruct logic for Expectations. To learn about other Expectation types, please see :ref:`Other Expectation decorators` below.
+
+
+4. **Load some data**
+
+    To make your new Expectations available for validation, you can instantiate a ``CustomSqlAlchemyDataset`` as follows:
+
+    .. code-block:: python
+
+        my_data_asset = CustomSqlAlchemyDataset(
+            "employees",
+            sa.create_engine("sqlite:///data/chinook.db")
+        )
+
+    If you have a Data Context configured, you can use ``DataContext.get_batch()`` to fetch a batch using a pre-configured Datasource. See :ref:`Configuring Datasources` and :ref:`Creating Batches` for instructions.
+
+    .. code-block:: python
+
+        import great_expectations as ge
+        context = ge.DataContext()
+
+        # You'll need to define this to create your batch:
+        my_batch_kwargs = ...
+
+        my_data_asset = context.get_batch(
+            my_batch_kwargs,
+            ExpectationSuite("my_temporary_test_suite"),
```

Comment:
This seems like a hack that is only useful in a guide since it will force the suite to be throwaway...in fact, I don't know if I've even tested that workflow...

Reply:
I tested it. It works.

I'm not deeply wedded to this approach, but I like the fact that the Expectation Suite is clearly a throwaway thing. It keeps this guide squarely focused on writing new custom Expectations, not building them into Suites.